lib/resourceutil/work.go

```diff
@@ -15,30 +15,31 @@ import (
 // labels[WorkPipelineNameLabel] = pipelineName
 // labels[WorkTypeLabel] = v1alpha1.WorkTypeResource
 
-func SetPromiseWorkLabels(l map[string]string, promiseName, pipelineName string) {
-	setWorkLabels(l, promiseName, "", pipelineName)
+func SetPromiseWorkLabels(l map[string]string, promiseName, pipelineName, workflowType string) {
+	setWorkLabels(l, promiseName, "", pipelineName, workflowType)
 }
 
-func SetResourceWorkLabels(l map[string]string, promiseName, resourceName, pipelineName string) {
-	setWorkLabels(l, promiseName, resourceName, pipelineName)
+func SetResourceWorkLabels(l map[string]string, promiseName, resourceName, pipelineName, workflowType string) {
+	setWorkLabels(l, promiseName, resourceName, pipelineName, string(v1alpha1.WorkTypeResource))
 }
 
 func SetStaticDependencyWorkLabels(l map[string]string, promiseName string) {
-	setWorkLabels(l, promiseName, "", "")
+	setWorkLabels(l, promiseName, "", "", string(v1alpha1.WorkTypeStaticDependency))
```

Comment:
```suggestion
	setWorkLabels(l, promiseName, "", "", v1alpha1.WorkTypeStaticDependency)
```

Nitpick: No need for conversion